src/category_theory/limits/shapes/split_coequalizer.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.limits.shapes.equalizers
+
+/-!
+# Split coequalizers
+
+We define what it means for a triple of morphisms `f g : X ⟶ Y`, `π : Y ⟶ Z` to be a split
+coequalizer: there is a section `s` of `π` and a section `t` of `g`, which additionally satisfy
+`t ≫ f = π ≫ s`.
+
+In addition, we show that every split coequalizer is a coequalizer
+(`category_theory.is_split_coequalizer.is_coequalizer`) and absolute
+(`category_theory.is_split_coequalizer.map`)
+
+A pair `f g : X ⟶ Y` is split if there is a `Z` and `π : Y ⟶ Z` making `f,g,π` a split coequalizer.
+A pair `f g : X ⟶ Y` is `G`-split if `G f, G g` is split.
+
+These definitions and constructions are useful in particular for the monadicity theorems.
+-/
+
+namespace category_theory
+
+universes v v₂ u u₂
+
+variables {C : Type u} [category.{v} C]
+variables {D : Type u₂} [category.{v} D]
+variables (G : C ⥤ D)
+variables {X Y : C} (f g : X ⟶ Y)
+
+/--
+A split coequalizer diagram consists of morphisms
+
+      f   π
+    X ⇉ Y → Z
+      g
+
+satisfying `f ≫ π = g ≫ π` together with morphisms
+
+      t   s
+    X ← Y ← Z
+
+satisfying `s ≫ π = 𝟙 Z`, `t ≫ g = 𝟙 Y` and `t ≫ f = π ≫ s`.
+
+The name "coequalizer" is appropriate, since any split coequalizer is a coequalizer, see
+`category_theory.is_split_coequalizer.is_coequalizer`.
+Split coequalizers are also absolute, since a functor preserves all the structure above.
+-/
+def is_split_coequalizer {Z : C} (π : Y ⟶ Z) : Prop :=
+f ≫ π = g ≫ π ∧ ∃ (s : Z ⟶ Y) (t : Y ⟶ X), s ≫ π = 𝟙 Z ∧ t ≫ g = 𝟙 Y ∧ t ≫ f = π ≫ s
```

Comment:
Doesn't it make more sense to turn this into a structure with several fields?
you are already choosing names below (where you write the accessor function).

Reply:
Yup, you're right.